scripts/release

```diff
@@ -102,6 +107,40 @@ parse_args_and_release() {
     stage=$(stage_from_version $new_version)
   fi
 
+  image_base="$(docker_repo_from_stage $stage)"
+  new_image="$image_base:$new_version"
+
+  # fail immediately if releasing to Github and either the username or token is not set
+  if [[ "$push" == "yes" ]] && [[ "$stage" == "final" ]] && [[ "$build_all" == "yes" || "$github_release" == "yes" || "$build_bundle" == "yes" || "$push_windows" == "yes" ]]; then
+    if [[ -z "$github_username" ]]; then
+      echo "Github username is required when releasing to Github." >&2
+      echo "Set the GITHUB_USERNAME env var or use the '--github-user <username>' option." >&2
+      exit 1
+    fi
+    if [[ -z "$github_token" ]]; then
+      echo "Github token is required when releasing to Github." >&2
+      echo "Set the GITHUB_TOKEN env var or use the '--github-token <token>' option." >&2
+      exit 1
+    fi
+  fi
+
+  # exit/fail immediately if only creating a Github release and the requirements are not met
+  if [[ "$github_release" == "yes" ]]; then
+    if [[ "$push" == "no" ]]; then
+      echo "Nothing to do"
+      exit 0
+    fi
+    if [[ "$stage" != "final" ]]; then
+      echo "Release stage must be 'final' in order to create a Github release." >&2
+      exit 1
+    fi
+    if ! docker pull $new_image; then
+      echo "$new_image must already be built and pushed in order to create a Github release." >&2
+      exit 1
+    fi
+    image_digest=$(get_image_digest $new_image)
```

Comment:
What is the advantage of doing this here and not just always waiting until right before creating the release?

Reply:
> What is the advantage of doing this here and not just always waiting until right before creating the release?

Leftover from testing. Will remove.